src/vaadin-overlay.html

```diff
@@ -578,7 +578,10 @@
 
       _enqueueAnimation(type, callback) {
         const handler = `__${type}Handler`;
-        const listener = () => {
+        const listener = event => {
+          if (event && event.target !== this) {
+            return;
+          }
```

Comment:
This part is most of the fix. This makes overlay ignore any unrelated `animationend` events that might be triggered by anything inside the dialog (e.g. if using `<vaadin-text-area>` inside the dialog which triggers a CSS animation on init).